analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs

```diff
@@ -56,11 +57,19 @@ public class SymbolicExecutionRunner : SymbolicExecutionRunnerBase
             new InitializationVectorShouldBeRandom(),
             new HashesShouldHaveUnpredictableSalt());
 
-        protected override ImmutableDictionary<DiagnosticDescriptor, RuleFactory> AllRules { get; } = ImmutableDictionary<DiagnosticDescriptor, RuleFactory>.Empty
+        public SymbolicExecutionRunner() : this(AnalyzerConfiguration.AlwaysEnabled)
+        {
+        }
+
+        internal /* for testing */ SymbolicExecutionRunner(IAnalyzerConfiguration configuration) : base(configuration)
+        {
+        }
+
+        protected override ImmutableDictionary<DiagnosticDescriptor, RuleFactory> RoslynRules { get; } = ImmutableDictionary<DiagnosticDescriptor, RuleFactory>.Empty
             .Add(RuleChecks.LocksReleasedAllPaths.S2222, CreateFactory<RuleChecks.LocksReleasedAllPaths>())
             .Add(RuleChecks.NullPointerDereference.S2259, CreateFactory<RuleChecks.NullPointerDereference>());
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => base.SupportedDiagnostics.Concat(SonarRules.SelectMany(x => x.SupportedDiagnostics)).ToImmutableArray();
+        protected override ImmutableArray<DiagnosticDescriptor> SonarRules => SonarAnalyzer.SelectMany(x => x.SupportedDiagnostics).ToImmutableArray();
```

Comment:
The SERunnerBase needs to be able to calculate "NonMigratedRules", which get executed in the "Roslyn" case.
Therefore the SERunnerBase needs to know about "new" and "old" rules and that is why "SonarRules" and "RoslynRules" are both exposed to the base now.